provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -905,6 +905,26 @@ func generateSchema() schema.PackageSpec {
 							"  - settings.kubernetes.cluster-dns-ip\n\n" +
 							"For an overview of the available settings, see https://bottlerocket.dev/en/os/1.20.x/api/settings/.",
 					},
+					"userData": {
+						TypeSpec: schema.TypeSpec{Type: "string"},
+						Description: "User specified code to run on node startup. This is expected to handle " +
+							"the full AWS EKS node bootstrapping. If omitted, the provider will configure the user data.\n\n" +
+							"See for more details: https://docs.aws.amazon.com/eks/latest/userguide/launch-templates.html#launch-template-user-data.",
+					},
+					"gpu": {
+						TypeSpec: schema.TypeSpec{Type: "boolean"},
+						Description: "Use the latest recommended EKS Optimized AMI with GPU support for the " +
+							"worker nodes.\nDefaults to false.\n\n" +
+							"Note: `gpu` and `amiId` are mutually exclusive.\n\n" +
+							"See for more details: https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-amis.html.",
+					},
+					"amiId": {
+						TypeSpec: schema.TypeSpec{Type: "string"},
+						Description: "The AMI ID to use for the worker nodes.\nDefaults to the latest recommended " +
```

Comment:
Likewise, should schema define `Default: "latest"` or some such?

Reply:
The default value here is not "latest". The provider performs actions in the background to fetch the latest recommended AMI ID.

Reply:
So it does so when amiId is empty/undefined, the provider does the search. OK, I guess the current form is the best.